### PR TITLE
dxDataGrid: Fix highlight searchText for the non string data types (T636268)

### PR DESCRIPTION
**Pull request by AntonSermyazhko**

### CODE REVIEW

File: js/ui/grid_core/ui.grid_core.search.js [hunk 1]
```diff
@@ -279,9 +280,11 @@ module.exports = {
                         $parent = cellElement.parent();
                         if(!$parent.length) {
                             $parent = $("<div>").append(cellElement);
+                        } else if(column) {
+                            findCondition = `td:nth-child(${column.index + 1})`;
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.search.js]: column.index is not a visible index of column.

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/grid_core/ui.grid_core.search.js (hunk 1 above)
Comment (AntonSermyazhko) [reply js/ui/grid_core/ui.grid_core.search.js]: Oh, sure. I've replaced index with visibleIndex.

File: js/ui/grid_core/ui.grid_core.search.js [hunk 2]
```diff
@@ -279,9 +280,11 @@ module.exports = {
                         $parent = cellElement.parent();
                         if(!$parent.length) {
                             $parent = $("<div>").append(cellElement);
+                        } else if(column) {
+                            findCondition = `td:nth-child(${column.visibleIndex + 1})`;
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.search.js]: This is also wrong

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/grid_core/ui.grid_core.search.js [hunk 3]
```diff
@@ -260,67 +260,86 @@ module.exports = {
...
                 },
 
-                _highlightSearchText: function(cellElement, isEquals, column) {
-                    var that = this,
-                        $parent,
-                        searchText = that.option("searchPanel.text");
-
-                    if(searchText && that.option("searchPanel.highlightSearchText")) {
-                        var normalizeString = that.option("searchPanel.highlightCaseSensitive") ? function(str) { return str; } : function(str) { return str.toLowerCase(); };
+                getFormattedSearchText: function(column, searchText) {
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.search.js]: Use "_" prefix for internal methods

File: js/ui/grid_core/ui.grid_core.search.js [hunk 4]
```diff
@@ -260,67 +260,86 @@ module.exports = {
...
                         }
+                    });
 
-                        $parent = cellElement.parent();
-                        if(!$parent.length) {
-                            $parent = $("<div>").append(cellElement);
+                    return $items;
+                },
+
+                processHighlightTextRecursive: function($content, searchText) {
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.search.js]: Strange name. May be "_highlightSearchTextCore" will be better.

**Review by vconst — CHANGES_REQUESTED** [review]

**Review by vconst — APPROVED** [review]

**Pull request merged**